include/z64collision_check.h

```diff
@@ -102,12 +152,12 @@ typedef struct {
 } ColliderJntSphElementDimInit; // size = 0x0C
 
 typedef struct {
-    /* 0x00 */ ColliderInfo info;
+    /* 0x00 */ ColliderElement base;
```

Comment:
The parent of `ColliderElement` for JntSph and other multi-elements colliders, is already an element, so this would mean `jntSphElem->elem` instead of the more natural (?) `jntSphElem->base`

and cylinders and other single-element colliders ofc already have `base` for the collider, and they don't have a specific element substruct, so their (basic) element is just "elem"

I could change the single-element colliders' elements to be specific structs, but it would just be bloat for no reason imo

Reply:
reason for calling Collider `base` in e.g. JntSph struct is because Colliders utilize a bit of polymorphism (e.g. SetAT accepts Collider* and downcasts to ColliderJntSph*). I don't think this is the case for ColliderElement. Additionally, I'd say it's a bit weird to only have some ColliderElement consumers to express polymorphism but not others.

Reply:
`ColliderElement` is at the start of every `Collider*Element`, and that naming pattern seems to justify the inheritance pattern (note the naming comes with the PR, `ColliderElement` was `ColliderInfo`-something)

See also https://github.com/zeldaret/oot/pull/1427#discussion_r1020958637 for why I think it makes sense to consider cylinders and such as single-elements colliders

and again
> I could change the single-element colliders' elements to be specific structs, but it would just be bloat for no reason imo

Reply:
but then you also continued saying 

> it would just be bloat for no reason imo

My position is you either implement ColliderShapeElements for all shapes, or you do it for none and change `base`->`elem`. There's no reason to go half and half here. Personally, I don't think the inheritance pattern is strictly necessary, which is why I didn't use it when decomping this code.

Reply:
I think it's a bad idea to make a change that would turn
```c
static ColliderCylinderInit sCylinderInit = {
    {
        COLTYPE_NONE,
        AT_NONE,
        AC_ON | AC_TYPE_ALL,
        OC1_ON | OC1_TYPE_ALL,
        OC2_TYPE_2,
        COLSHAPE_CYLINDER,
    },
    {
        ELEMTYPE_UNK2,
        { 0x00000000, 0x00, 0x00 },
        { 0xFFCFFFFF, 0x00, 0x00 },
        TOUCH_NONE,
        BUMP_ON,
        OCELEM_ON,
    },
    { 25, 60, 0, { 0, 0, 0 } },
};
```
into
```c
static ColliderCylinderInit sCylinderInit = {
    {
        COLTYPE_NONE,
        AT_NONE,
        AC_ON | AC_TYPE_ALL,
        OC1_ON | OC1_TYPE_ALL,
        OC2_TYPE_2,
        COLSHAPE_CYLINDER,
    },
    {
        {
            ELEMTYPE_UNK2,
            { 0x00000000, 0x00, 0x00 },
            { 0xFFCFFFFF, 0x00, 0x00 },
            TOUCH_NONE,
            BUMP_ON,
            OCELEM_ON,
        },
        { 25, 60, 0, { 0, 0, 0 } },
    },
};
```
or (if one comma is omitted):
```c
static ColliderCylinderInit sCylinderInit = {
    {
        COLTYPE_NONE,
        AT_NONE,
        AC_ON | AC_TYPE_ALL,
        OC1_ON | OC1_TYPE_ALL,
        OC2_TYPE_2,
        COLSHAPE_CYLINDER,
    },
    { {
          ELEMTYPE_UNK2,
          { 0x00000000, 0x00, 0x00 },
          { 0xFFCFFFFF, 0x00, 0x00 },
          TOUCH_NONE,
          BUMP_ON,
          OCELEM_ON,
      },
      { 25, 60, 0, { 0, 0, 0 } } },
};
```

(note these are the Init structs but I assume we want to keep them mirroring the live structs)

even though it would fit the pattern of elements being `ColliderElement base;` followed by element dimensions

Or do you really think it's worth it

Reply:
well... that's the price you pay for consistency. I do hate it makes var access a little longer, but you end up with this:

    cylinder->element.base
    cylinder->element.dim
    jntSphr->elements[0].base
    jntSphr->elements[0].dim

Reply:
Sticking to `ColliderJntSph.elements[].base` and `ColliderCylinder.elem`, at least for now.